packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,685 @@
+        ],
+      }),
+    );
+    const provider = new cr.Provider(this, 'CustomResourceProvider', {
+      onEventHandler: shellCliLambda,
+    });
+    new cdk.CustomResource(this, 'Custom Resource', {
+      serviceToken: provider.serviceToken,
+    });
+  }
+
+  private createPolicyStatements(): iam.PolicyStatement[] {
+    const policyStatements = [
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+            service: 'emr-containers',
+            resource: 'virtualclusters',
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualCluster.id) ? '*' : this.props.virtualCluster.id, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+          }),
+        ],
+        actions: ['emr-containers:StartJobRun'],
+        conditions: {
+          StringEquals: {
+            'emr-containers:ExecutionRoleArn': this.role.roleArn,
+          },
+        },
+      }),
+    ];
+
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+              service: 'emr-containers',
+              resource: 'virtualclusters',
+              resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualCluster.id) ? '*' : `${this.props.virtualCluster.id}/jobruns/*`, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+            }),
+          ],
+          actions: [
+            'emr-containers:DescribeJobRun',
+            'emr-containers:CancelJobRun',
+          ],
+        }),
+      );
+    }
+
+    return policyStatements;
+  }
+}
+
+/**
+ * The information about job driver for Spark submit.
+ */
+export interface SparkSubmitJobDriver {
+
+  /**
+   * The entry point of job application.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 256.
+   */
+  readonly entryPoint: sfn.TaskInput;
+
+  /**
+   * The arguments for a job application in a task input object containing an array of strings
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 10280.
+   * @type string[]
+   *
+   * @default - No arguments defined
+   */
+  readonly entryPointArguments?: sfn.TaskInput;
+
+  /**
+   * The Spark submit parameters that are used for job runs.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 102400.
+   *
+   * @default - No spark submit parameters
+   */
+  readonly sparkSubmitParameters?: string;
+}
+
+/**
+ * Specify the driver that the EMR Containers job runs on.
+ * The job driver is used to provide an input for the job that will be run.
+ */
+export interface JobDriver {
+
```

Comment:
```suggestion
```

I know these are all small nits (sorry about that) but I would vastly appreciate a consistent syntax and sometimes you have a space here (i.e. `JobDriver` and `Classification`) but then other times you don't (i.e. `ApplicationConfiguration`). I would prefer you don't have spaces here, but what matters most to me is that its consistent.

Reply:
Removed spaces from all the places to ensure consistency